### ISSUE TITLE
aria-valuenow should be optional for role=spinbutton

### ISSUE THREAD

**Issue by devongovett** [dequelabs/axe-core#3505]
Product: axe-core

Expectation: the aria spec says:

> Authors SHOULD set the [aria-valuenow](https://w3c.github.io/aria/#aria-valuenow) attribute when the [spinbutton](https://w3c.github.io/aria/#spinbutton) has a value.

This allows `aria-valuenow` to be omitted when there is no value set (e.g. a placeholder is shown, or the field is empty).

Actual: aXe errors with "Required ARIA attribute not present: aria-valuenow".

Motivation: Building a date picker where a month field might display "mm" when a value hasn't been entered yet.

<pre><code>
axe-core version: 4.4.2

Browser and Assistive Technology versions

For Tooling issues:
- Node version: 16.13.1
- Platform:  Mac
</code></pre>

**Comment by straker** [comment]
Thanks for the issue. Looks like we're still using the ARIA 1.1 spec which requires `aria-valuenow` so we just need to update to the new 1.2 spec.

**Comment by padmavemulapati** [comment]
Validated with the latest develop branch code base, 
aria-required-attr` is passing when attribute is `role=spinbutton` and failing when `role=heading` or `role=checkbox`

<img width="1306" alt="image" src="https://user-images.githubusercontent.com/47409579/187351056-df3de28b-acef-474c-ba0b-f35a6dff433c.png">
